### PR TITLE
New Package: Brainfunk-0.9.0

### PR DESCRIPTION
**Pull request by misuchiru03**

### CODE REVIEW

File: srcpkgs/png++-devel/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'png++-devel'
+pkgname=png++-devel
+version=0.2.9
+revision=1
+noarch=yes
+_pkgname=png++
```
Comment (Vaelatern) [inline srcpkgs/png++-devel/template]: Maybe you want `libpng++` instead of this dance? Or even just `png++`.

File: srcpkgs/png++/template [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+build_style=gnu-makefile
+make_build_target="docs"
+make_use_env=yes
+hostmakedepends="doxygen"
+makedepends="libpng-devel"
+short_desc="C++ wrapper for linpng library"
+maintainer="Alex Childs <misuchiru03+void@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://www.nongnu.org/pngpp"
+distfiles="${NONGNU_SITE}/releases/pngpp/${pkgname}-${version}.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/png++/template]: You don't need two templates for the same thing.

File: srcpkgs/png++-devel/template (hunk 1 above)
Comment (misuchiru03) [reply srcpkgs/png++-devel/template]: I saw the issue here.  The package name of the `png++-devel` is supposed to be `png++`.  I'm fixing it right now.

File: srcpkgs/png++/template (hunk 2 above)
Comment (misuchiru03) [reply srcpkgs/png++/template]: Right. I had added `png++{,-doc}` and failed to remove the older `png++-devel{,-doc}`

File: srcpkgs/png++/template [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'png++'
+pkgname=png++
+version=0.2.9
+revision=1
+noarch=yes
+build_style=gnu-makefile
+make_build_target="docs"
```
Comment (Vaelatern) [inline srcpkgs/png++/template]: Just docs? Are you sure this actually installs a binary?
Pretty sure you want another target here.

**Comment by Vaelatern** [comment]
Is this actually a noarch package?

File: srcpkgs/png++/template (hunk 3 above)
Comment (misuchiru03) [reply srcpkgs/png++/template]: for reference: https://github.com/void-linux/void-packages/pull/3894

File: srcpkgs/Brainfunk/template [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'Brainfunk'
+pkgname=Brainfunk
+version=0.9.0
+revision=1
+wrksrc=${pkgname}-${version}-beta
+build_style=gnu-makefile
+make_use_env=yes
+hostmakedepends="png++"
```
Comment (Vaelatern) [inline srcpkgs/Brainfunk/template]: Suspect you want `makedepends` here.

Comment (misuchiru03) [reply srcpkgs/Brainfunk/template]: What I just noticed as well, before checking this.  Makes more sense, and it should work.

**Comment by Vaelatern** [comment]
Ok, now cross works, but `-musl` doesn't yet.

**Comment by misuchiru03** [comment]
Right. Looks like it will need some heavy patching as 1-2 args are missing when passing to musl's compilers for several files. Will keep on keeping on. Marking as WIP for now

**Comment by Johnnynator** [comment]
This should get `png++` working on musl
```
--- error.hpp	2015-10-25 20:42:45.000000000 +0100
+++ -	2018-11-07 12:49:36.093450988 +0100
@@ -100,7 +100,7 @@
             strerror_s(buf, ERRBUF_SIZE, errnum);
             return std::string(buf);
 #else
-#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE
+#if ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE) || !defined(__GLIBC__)
             strerror_r(errnum, buf, ERRBUF_SIZE);
             return std::string(buf);
 #else
--- config.hpp	2015-10-25 20:42:45.000000000 +0100
+++ -	2018-11-07 12:52:38.903453133 +0100
@@ -34,7 +34,7 @@
 #include <stdlib.h>
 
 // Endianness test
-#if defined(__GLIBC__)
+#if defined(__linux__)
 
 #include <endian.h>
```

**Pull request closed without merging**